File: web/src/api/task.js
import service from "@/utils/request";

export const getTaskList = (params) => {
    return service({
        url: "/task/getTaskList",
        method: "get",
        params
    })
}

export const createTask = (data) => {
    return service({
        url: "/task/createTask",
        method: "post",
        data
    })
}

export const switchTaskStatus = (data) => {
    return service({
        url: "/task/switchTaskStatus",
        method: "post",
        data
    })
}

File: server/router/task.go
package router

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/madneal/gshark/api/v1"
	"github.com/madneal/gshark/middleware"
)

func InitTaskRouter(Router *gin.RouterGroup) {
	TaskRouter := Router.Group("task").Use(middleware.OperationRecord())
	{
		TaskRouter.GET("getTaskList", v1.GetTaskList)
		TaskRouter.POST("createTask", v1.CreateTask)
		TaskRouter.POST("switchTaskStatus", v1.SwitchTaskStatus)
	}
}
